src/server/game/Movement/MovementGenerators/FlightPathMovementGenerator.cpp

```diff
@@ -151,6 +151,8 @@ void FlightPathMovementGenerator::DoFinalize(Player* owner, bool active, bool/*
         // when client side flight end early in comparison server side
         owner->StopMoving();
         owner->SetFallInformation(0, owner->GetPositionZ());
+        // When the player reaches the last flight point, teleport to destination at floor Z
+        owner->TeleportTo(_path[GetCurrentNode()]->MapID, _path[GetCurrentNode()]->LocX, _path[GetCurrentNode()]->LocY, owner->GetFloorZ(), owner->GetOrientation());
```

Comment:
Not sure I like the fact that we use X and Y of the path but Floor Z of the Player (which could be different than Floor Z of the path)

Reply:
Also I would ask for a sniff validation for this behaivior because for me this smells like a hack tbh.

Reply:
All the cases I have seen (videos) cause a teleport exactly on the ground, so I went with it as it made sense to me.

Could use the endpoint's FloorZ instead, they should be the same.

Reply:
Videos do not necessarily validate this solution. You need to search for a teleport packet inside a sniff for the behaivior. If it does not exist, the problem is located somewhere else which is probably going to be the flight path spline movement.

Reply:
Confirmed with sniffs, client sends MSG_MOVE_TELEPORT_ACK at the end of the flight. The endpoint serverside is higher on the ground, but teleports instantly instead of falling.

This is the flow:
```
Taxi creature finished moving to endpoint: SMSG_ON_MONSTER_MOVE: X: 2793.49 Y: 909.4843 Z: 24.54242 (Northrend)
Control is given to player: SMSG_CONTROL_UPDATE
Collision height is reset because the player dismounted: SMSG_MOVE_SET_COLLISION_HEIGHT
Teleport. MSG_MOVE_TELEPORT Destination: X: 2792.45 Y: 908.96 Z: 22.33 (FloorZ pos)
```